zilliqa/src/exec.rs

```diff
@@ -644,6 +647,26 @@ impl State {
         Ok(())
     }
 
+    pub fn leader_at_block(&self, block: &Block) -> Result<NodePublicKey> {
+        let data = contracts::deposit::LEADER.encode_input(&[])?;
```

Comment:
After looking at `Consensus::leader_at_block()` I trust this returns the leader based on the state of the `block` argument. I'm just wondering why `get_stakers_at_block()` uses a different way to return the stakers based on a given block's state.